server/src/data/stores/distinct.js

```diff
@@ -0,0 +1,135 @@
+'use strict';
+
+const bluebird = require('bluebird');
+const moment = require('moment');
+const winston = require('winston');
+
+const DEFAULT_EXPIRATION_TIME = moment.duration(1, 'week');
+const lastElement = arr => arr[arr.length - 1];
+
+const inMemoryStore = (options) => {
+    let keyExpiration = (options || {}).keyExpiration;
+    if (keyExpiration === undefined) {
+        keyExpiration = DEFAULT_EXPIRATION_TIME;
+    }
+
+    const store = new Map();
+
+    const cleanup = () => {
+        if (!keyExpiration) {
+            return;
+        }
+        const keysExpirationTime = moment().subtract(keyExpiration);
+        [...store.keys()].forEach(key => {
+            const keyAdditionTime = store.get(key);
+            if (keyAdditionTime < keysExpirationTime) {
+                store.delete(key);
+            }
+        });
+    };
+
+    // TODO: find a way to use freeze here and not break tests
+    return Object.seal({
+        add(item) {
+            store.set(item, moment());
+            return Promise.resolve();
+        },
+        get size() {
+            cleanup();
+            return Promise.resolve(store.size);
+        },
+        get items() {
+            cleanup();
+            return Promise.resolve([...store.keys()]);
+        }
+    });
+};
+
+const redisStore = (options) => {
```

Comment:
Is there a reason why you didn't reuse the existing redis wrapper in `data/stores/redis`?